src/libraries/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509ChainElement.cs

```diff
@@ -15,14 +15,14 @@ namespace System.Security.Cryptography.X509Certificates
 {
     public class X509ChainElement
     {
-        internal X509ChainElement(X509Certificate2 certificate, X509ChainStatus[] chainElementStatus, string information)
+        internal X509ChainElement(X509Certificate2? certificate, X509ChainStatus[] chainElementStatus, string information)
```

Comment:
When is this ever passed null?

Reply:
In the macOS PAL https://github.com/dotnet/runtime/blob/aac8f91cc5510b3d271d86eef616b6007b6e6631/src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.OSX/ChainPal.cs#L365

Reply:
I'd go ahead and mark it as non-nullable, and BuildElement, and bang it away on this line (unless you want to break apart BuildElement into BuildElement and BuildElementStatus... but I'd vote for just the bang here, maybe with a comment that the null-certificate element is never returned).

Reply:
Done.